src/scripts.js

```diff
@@ -96,15 +125,34 @@ function cardButtonConditionals(event) {
   if (event.target.classList.contains('favorite')) {
     favoriteCard(event);
   }
-  if (event.target.classList.contains('card-picture')) {
+  if (event.target.classList.contains('instruction')) {
+    declareRecipe(event)
     findDirections(event);
   } 
-  if(event.target.classList.contains('add-button')){
-    // console.log(typeof event.target.id)
+  if(event.target.classList.contains('add-button')) {
+    console.log(event)
     addToCookLater(event);
   }
- }
-
+  if(event.target.classList.contains('icon')) {
+    let id = event.target.id
+    let currentRecipe = cookbook.recipes.find(x => x.id === +id)
+    let ingredientFeedback = Object.values(userPantry.giveFeedbackOnIngredients(currentRecipe, cookbook))
+    domUpdates.displayIngredientFeedback(ingredientFeedback, id)
+  }
+  if(event.target.classList.contains('back-button')){
+    domUpdates.populateCards(cookbook.recipes, cardArea, currentFavs())
+  }
+  if(event.target.classList.contains('purchase-button')){
+    let id = event.target.id
+    const recipe = declareRecipe(event)
+    const post = postIngredients(recipe) 
+    post.then(values => Promise.all(values))
+    .then(data => data.map(response => response.json()))
```

Comment:
I think line 149 is redundant here, but i'm not totally sure. The return value of `postIngredients`, as seen on line 91 of scripts, is `return Promise.all(promisesReturned)`. 

So think the Promise.all on 149 is essentially a duplicate, but i could be wrong. I guess try taking it out and just chaining straight to 150, with `post.then(etc.etc.)` and let me know if that works.